packages/template/vite/tmpl/renderer.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * This file will automatically be loaded by vite and run in the "renderer" context.
+ * To learn more about the differences between the "main" and the "renderer" context in
+ * Electron, visit:
+ *
+ * https://electronjs.org/docs/tutorial/application-architecture#main-and-renderer-processes
+ *
+ * By default, Node.js integration in this file is disabled. When enabling Node.js integration
+ * in a renderer process, please be aware of potential security implications. You can read
+ * more about security risks here:
+ *
+ * https://electronjs.org/docs/tutorial/security
```

Comment:
More of a Vite question than specific to this template: will the Vite template still work with node integration disabled? I remember that being an issue in the past?

Reply:
Vite works fine with node disabled, without any potential issues. Because it always works with the `vite build` command, which will build a Boundle using Rollup, which is always compatible with CJS and ESM modules. Perfect like Webpack! 🚀

By the way, the problem with Vite's integration with Electron arises when using node in the render process. Because `vite serve` is forced to rely on ESM format modules, which will cause many npm-packages of node to not work well, so I developed `vite-plugin-electron-renderer` to solve this problem. But having said that, our @electron-forge/template-vite template does not provide node integration in the rendering process by default, but it is up to the user to decide which solution to use. 🤔

All in all, our templates can be used instead of Webpack templates without any risk. 😄

Reply:
@caoxiemeihao Would it be possible to have a mixed environment where we enable `vite-plugin-electron-renderer` only for certain renderers but not for others?

Reply:
@erickzhao If the user has a configuration with multiple rendering processes, such as
`vite.renderer1.config.js`
`vite.renderer2.config.js`
they can be introduced at the user's discretion `vite-plugin-electron-renderer` or not.

If the user has only one configuration file `vite-plugin-electron-renderer` does not accurately distinguish between a renderer that needs to integrate with node
However, I can develop this feature. It looks like this
```js
// By automatically adding the querystring of nodeIntegration to distinguish that some renderers need to integrate node.
import { readFile } from 'fs?nodeIntegration'
```

Reply:
> because Vite always builds all files with the same configuration

I'm not super familiar with Vite and Rollup configuration, but would it be possible to use an array for RollupOptions?

https://github.com/rollup/rollup/issues/863#issuecomment-306061779

Reply:
Now, Vite can't support use an array https://github.com/vitejs/vite/issues/2039
What if we redesigned the API? What do you think?

```diff
{
-  CLIOptions: VitePluginRendererConfig,
+  renderer: VitePluginRendererConfig[],
}
```

If we support a Renderer array config, this will result in multiple Vite servers being started at the same time during development, **e.g.**
- foo.html - `localhost:5173`
- bar.html - `localhost:5174`

Reply:
I think having the option to do that would be useful 👍

Reply:
> I think having the option to do that would be useful 👍

feat!:support multiple renderer 7a3bd0a

Reply:
> I think having the option to do that would be useful 👍

feat!:support multiple renderer 7a3bd0a